File: Java version/app/src/test/java/it/unical/mat/embasp/specializations/dlv2/From.java
package it.unical.mat.embasp.specializations.dlv2;

import it.unical.mat.embasp.languages.Id;
import it.unical.mat.embasp.languages.Param;

@Id("from")
public class From {

	@Param(0)
	private int x;
	
	public From() {
		this.x = 0;
	}
	
	public From(int x) {
		this.x = x;
	}
	
	public int getX() {
		return x;
	}
	
	public void setX(int x) {
		this.x = x;
	}
	
}

File: CSharp version/it/unical/mat/test/specialization/dlv2/From.cs
using it.unical.mat.embasp.languages;
using System;
using System.Collections.Generic;
using System.Text;

namespace it.unical.mat.test
{
    [Id("from")]
    class From
    {
        [Param(0)]
        private int x;

        public From()
        {
            this.x = 0;
        }

        public From(int x)
        {
            this.x = x;
        }

        public int getX()
        {
            return x;
        }

        public void setX(int x)
        {
            this.x = x;
        }
    }
}
